### PR TITLE
DATAUP-532: enable batch queries in the frontend

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

The backend will soon be able to accept info/status/start updates/stop updates requests for all jobs in a batch, so this PR puts the frontend machinery in the `jobCommChannel.js` module in place.

I also made the FE job request naming conventions more uniform.

- Adding new batch queries into the job comm channel
- Converting "request-job-cancellation" => "request-job-cancel" for a more uniform API experience
- Updating job management docs

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-532
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: docs/design/job_architecture.md [hunk 1]
```diff
@@ -21,28 +21,27 @@ All the `request-job-*` requests take as arguments either a single job ID string
...
 
 `request-job-info` - request information about the job(s), specifically app id, spec, input parameters and (if finished) outputs
   * `jobId` - a string, the job id OR
-  * `jobIdList` - an array of job IDs
-  * `parentJobId` - (optional) a string, the id of the requested job's "parent" job
+  * `jobIdList` - an array of job IDs OR
+  * `batchId` - a batch parent (make the request for all jobs in the batch)
 
-`request-job-cancellation` - request that the server cancel the running job(s)
+`request-job-cancel` - request that the server cancel the running job(s)
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:42]: made this more uniform -- who cares if it isn't perfect English?!

File: docs/design/job_architecture.md [hunk 2]
```diff
@@ -214,35 +213,29 @@ These are organized by the `request_type` field, followed by the expected respon
...
+* `job_id` - string OR `job_id_list` - array of strings OR `batch_id` - string
 
 `job_info` - request general information about job(s), responds with `job_info` for each job
-* `job_id` - string OR `job_id_list` - array of strings
-* `parent_job_id` - optional string
+* `job_id` - string OR `job_id_list` - array of strings OR `batch_id` - string
 
 `job_logs` - request job log information, responds with `job_logs` for each job
 * `job_id` - string OR `job_id_list` - array of strings
-* `parent_job_id` - optional string
```
Comment (ialarmedalien) [inline docs/design/job_architecture.md:238]: removed all these references to `parent_job_id`, which are no longer relevant

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 3]
```diff
@@ -56,40 +56,51 @@ define([
...
-        // ensures that job polling is running.
-        'request-job-updates-start': START_JOB_UPDATE,
-        // Tells kernel to stop including a job in the lookup loop.
-        'request-job-updates-stop': STOP_JOB_UPDATE,
-
         // cancels the job
-        'request-job-cancellation': CANCEL_JOB,
-        // retries the job
-        'request-job-retry': RETRY_JOB,
+        'request-job-cancel': JOB_REQUESTS.CANCEL,
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:85]: reordered these lines

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 4]
```diff
@@ -351,6 +371,26 @@ define([
                     });
                     break;
 
+                case 'jobs_retried':
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:374]: prelim retry handling

I have tests for this in a branch somewhere but cannot find them at the moment... 😖

File: test/unit/spec/common/jobCommChannel-spec.js [hunk 5]
```diff
@@ -130,27 +130,6 @@ define(['common/jobCommChannel', 'base/js/namespace', 'common/runtime', 'testUti
                 message: { jobId: 'someJob' },
                 expected: { request_type: 'retry_job', job_id: 'someJob' },
             },
-            {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobCommChannel-spec.js:133]: abstract out these tests -- see below

File: kbase-extension/static/kbase/js/common/jobCommChannel.js [hunk 6]
```diff
@@ -56,40 +56,51 @@ define([
...
-        START_JOB_UPDATE = 'start_job_update',
-        CANCEL_JOB = 'cancel_job',
-        RETRY_JOB = 'retry_job',
-        JOB_LOGS = 'job_logs',
-        JOB_INFO = 'job_info',
+        CELL = 'cell',
         JOB = 'jobId',
-        CELL = 'cell';
+        // messages to be sent to backend for job request types
+        JOB_REQUESTS = {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/jobCommChannel.js:62]: 👍  easier to read!

File: kbase-extension/static/kbase/js/common/jobCommChannel.js (hunk 4 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/common/jobCommChannel.js:374]: 🤷‍♂️

**Review by briehl — APPROVED** [review]
Seems fine. I don't think anything calls the `_batch` version of the job comm messages yet, right? At least, I didn't see it in the other PR.

So, next step is to merge in #2403 with the batch BE changes, then wire this up to `jobmanager.js` (and other places)?

**Comment by briehl** [comment]
Unit tests blipped, so I restarted them. Hopefully just a fluke, as it's in an unrelated module.

**Pull request merged**